cmd/geth/chaincmd.go

```diff
@@ -270,61 +257,140 @@ func initNetwork(ctx *cli.Context) error {
 			ips[i] = "127.0.0.1"
 		}
 	}
+	return ips, nil
+}
+
+func createPorts(ipStr string, port int, size int) []int {
+	ports := make([]int, size)
+	if len(ipStr) == 0 { // localhost , so different ports
+		for i := 0; i < size; i++ {
+			ports[i] = port + i
+		}
+	} else { // different machines, keep same port
+		for i := 0; i < size; i++ {
+			ports[i] = port
+		}
+	}
+	return ports
+}
+
+// Create config for node i in the cluster
+func createNodeConfig(baseConfig gethConfig, enodes []*enode.Node, ip string, port int, size int, i int) gethConfig {
+	baseConfig.Node.HTTPHost = ip
+	baseConfig.Node.P2P.ListenAddr = fmt.Sprintf(":%d", port+i)
+	baseConfig.Node.P2P.BootstrapNodes = make([]*enode.Node, size-1)
+	// Set the P2P connections between this node and the other nodes
+	for j := 0; j < i; j++ {
+		baseConfig.Node.P2P.BootstrapNodes[j] = enodes[j]
+	}
+	for j := i + 1; j < size; j++ {
```

Comment:
Should this be `size` or `size+1` ?

Reply:
Is size of `enodes` and size of `baseConfig.Node.P2P.BootstrapNodes` supposed to be the same?

Reply:
No, `baseConfig.Node.P2P.BootstrapNodes` length is `size-` while `enodes` length is `size`. This is because the node does not include itself in the list of bootstrap nodes (i.e. only the other nodes).